gensym/ulid.go

```diff
@@ -1,30 +1,30 @@
-package server
+package gensym
 
 import (
 	"context"
-	"errors"
 	"math/rand"
-	"strings"
 	"sync"
 
 	"github.com/deref/exo/chrono"
-	ulid "github.com/oklog/ulid/v2"
+	"github.com/oklog/ulid/v2"
 )
 
-type idGen struct {
+// A concurrency-safe source of monotonic ULIDs.
+// See <https://github.com/ulid/spec>.
+type ULIDGenerator struct {
 	mu      sync.Mutex
 	entropy *ulid.MonotonicEntropy
 }
 
-func newIdGen(ctx context.Context) *idGen {
+func NewULIDGenerator(ctx context.Context) *ULIDGenerator {
 	seed := int64(chrono.NowNano(ctx))
 	randRead := rand.New(rand.NewSource(seed))
-	return &idGen{
+	return &ULIDGenerator{
 		entropy: ulid.Monotonic(randRead, 0),
 	}
 }
 
-func (gen *idGen) nextId(ctx context.Context) ([]byte, error) {
+func (gen *ULIDGenerator) NextID(ctx context.Context) ([]byte, error) {
 	gen.mu.Lock()
```

Comment:
Oh, btw, this lock is in here because the math/rand source is not thread-safe :(